controllers/nodeupgrade_controller.go

```diff
@@ -384,8 +385,32 @@ func upgraderPodExists(ctx context.Context, remoteClient client.Client, nodeName
 
 func getUpgraderPod(ctx context.Context, remoteClient client.Client, nodeName string) (*corev1.Pod, error) {
 	pod := &corev1.Pod{}
-	if err := remoteClient.Get(ctx, getNamespacedNameType(upgrader.PodName(nodeName), constants.EksaSystemNamespace), pod); err != nil {
+	if err := remoteClient.Get(ctx, GetNamespacedNameType(upgrader.PodName(nodeName), constants.EksaSystemNamespace), pod); err != nil {
 		return nil, err
 	}
 	return pod, nil
 }
+
+// nodeUpgradeName returns the name of the node upgrade object based on the machine reference.
+func nodeUpgraderName(machineRefName string) string {
+	return fmt.Sprintf("%s-node-upgrade", machineRefName)
+}
+
+func nodeUpgrader(machineRef anywherev1.Ref, kubernetesVersion, etcdVersion string, firstControlPlane bool) *anywherev1.NodeUpgrade {
+	return &anywherev1.NodeUpgrade{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      nodeUpgraderName(machineRef.Name),
+			Namespace: constants.EksaSystemNamespace,
+		},
+		Spec: anywherev1.NodeUpgradeSpec{
+			Machine: corev1.ObjectReference{
+				Kind:      machineRef.Kind,
+				Namespace: constants.EksaSystemNamespace,
+				Name:      machineRef.Name,
+			},
+			KubernetesVersion:     kubernetesVersion,
+			EtcdVersion:           &etcdVersion,
+			FirstNodeToBeUpgraded: firstControlPlane,
```

Comment:
Wondering if we want to separate cp and worker node as a field here instead of saying it's not the first node to be upgraded, but that also means the upgrader pod needs to take in that information.

Reply:
Yeah these fields are ignored in the md upgrader pods and aren't required technically so I could separate them

Reply:
Can we at least make this field optional if we pass it in from md controller? But if it's too much refactor, then I say just leave it for now until we figure it out later.